packages/task/src/browser/quick-open-task.ts

```diff
@@ -196,25 +199,39 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
     attach(): void {
         this.items = [];
         this.actionProvider = undefined;
-
+        const isMulti: boolean = this.workspaceService.isMultiRootWorkspaceOpened;
         this.taskService.getRunningTasks().then(tasks => {
             if (!tasks.length) {
                 this.items.push(new QuickOpenItem({
                     label: 'No tasks found',
                     run: (_mode: QuickOpenMode): boolean => false
                 }));
-            }
-            for (const task of tasks) {
-                // can only attach to terminal processes, so only list those
-                if (task.terminalId) {
-                    this.items.push(
-                        new TaskAttachQuickOpenItem(
+            } else {
+                tasks.forEach((task: TaskInfo) => {
+                    // can only attach to terminal processes, so only list those
+                    if (task.terminalId) {
+                        this.items.push(new RunningTaskQuickOpenItem(
                             task,
-                            this.getRunningTaskLabel(task),
-                            this.taskService
-                        )
-                    );
-                }
+                            this.taskService,
+                            this.taskNameResolver,
+                            this.taskSourceResolver,
+                            this.taskDefinitionRegistry,
+                            this.labelProvider,
+                            isMulti,
+                            {
+                                run: (mode: QuickOpenMode): boolean => {
+                                    if (mode !== QuickOpenMode.OPEN) {
+                                        return false;
+                                    }
+                                    if (task.terminalId) {
+                                        this.taskService.attach(task.terminalId, task.taskId);
+                                    }
```

Comment:
btw, the question is **not** related to your PR, but maybe you have an answer: 
To be honest, I didn't use this action and I don't see `Terminal` -> `Attach Task` action in VS Code. 
Tried to use the action in Theia:
 - `quick-open` items are created for running tasks with `terminalId`, so, I guess, a terminal widget already open for such tasks 
-  it creates one more terminal for a running task
What's the use case for using this action?  

thanks in advance!

Reply:
It is used in 2 cases in my opinion
1. the user closed the browser for some reason, when s/he reopens the workspace s/he wants to reconnect to the task
2. the user wants someone that opens the same workspace to see the task output

Reply:
Why I asked:
it's strange for me that it creates one more terminal: 
- we give `terminalId`, `taskId`
- `attach` method asks  `widgetManager` **get** or create widget.
So, it's expected to get existed widget, but it creates a new one.

OK, anyway it's not related to the current PR. 
thank you for the answer!

Reply:
> Why I asked:
> it's strange for me that it creates one more terminal:
> 
> * we give `terminalId`, `taskId`
> * `attach` method asks  `widgetManager` **get** or create widget.
> 
> So, it's expected to get existed widget, but it creates a new one.

yea i filed an issue for the problem you describe: https://github.com/eclipse-theia/theia/issues/7407

i think:
1. we should set focus to the teminal widget if there is already a terminal widget
2. we should create a new widget if we couldn't find that widget - this is likely to happen, as the localstorage could possibly be cleared, or users use a different machine to access the same workspace